src/region.js

```diff
@@ -204,12 +208,38 @@ Marionette.Region = Marionette.Object.extend({
     return Backbone.$(el, Marionette._getValue(this.options.parentEl, this));
   },
 
+  // Replace the region's DOM element with the view's DOM element.
+  replaceEl: function(view) {
+    // empty el so we don't save any non-destroyed views
+    this.$el.contents().detach();
```

Comment:
yeah thats the thing with `preventDestroy` if that is used Mn has no power over memory leaks, personally I hate the option (views should be recreated) but it is used quite a bit.

I think for 2.5 we'll have to stick with `detach` here. For 3.0 im open to removing the `preventDestroy` function but that needs some discussion. Ill open an issue